### PR TITLE
Add note about failing upgrade tests

### PR DESCRIPTION
**Pull request by VenelinMartinov**
Add a short note about fixing failing upgrade tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,12 @@ The resulting recorded snapshot files are currently expected to be checked into
 used to inform `upgrade-quick` and `upgrade-preview-only` tests. When updating the baseline version,
 snapshots need to be recorded anew on the new version.
 
+### Fixing failing tests
+If a test fails after changing it then the snapshot files likely need to be re-recorded.
```
Comment (t0yv0) [inline README.md:89]: Oof, I think this statement "If a test fails after changing it then the snapshot files likely need to be re-recorded." is exactly false, but perhaps we're mis-communicating one more time. Can you elaborate? When does a failing test lead to the need to re-record baselines..

**Comment by t0yv0** [comment]
Should we close @VenelinMartinov or still editing this?

**Comment by VenelinMartinov** [comment]
I think I forgot to answer here, does this version make sense?

I think this is what I meant by "the test changes"

**Comment by t0yv0** [comment]
You have a good point that any time source changes are made to the program under test, the snapshots need to be recorded again. We could enforce that by a checksum. So you would never forget to record anew.

The verbiage in the PR is still a little confusing to me though.

**Comment by t0yv0** [comment]
Something like this?

```

** Fixing failing tests

- If the tests fail by flagging unwanted resource updates or replacements that are actually
  acceptable, configure or custom
  [DiffValidation](https://github.com/pulumi/providertest/blob/5f23c3ec7cee882392ea356a54c0f74f56b0f7d5/upgrade.go#L241)
  setting with more relaxed asserts.

- If the tests flag legitimate upgrade issues, fixes are necessarily specific to the provider and resource being tested. 

- Remember to re-record the test snapshots when making changes to the example program or the
  baseline provider dependency.

```

**Comment by VenelinMartinov** [comment]
Thanks for clearing that up, that looks very sensible!

**Review by t0yv0 — APPROVED** [review]

**Pull request merged**